repondeur/zam_repondeur/models/lecture.py

```diff
@@ -187,23 +195,28 @@ def get(
 
     @classmethod
     def exists(
-        cls, chambre: Chambre, texte: "Texte", partie: Optional[int], organe: str
+        cls,
+        dossier: "Dossier",
+        texte: "Texte",
+        partie: Optional[int],
+        phase: Phase,
+        chambre: Chambre,
+        organe: str,
     ) -> bool:
-        res: bool = DBSession.query(
-            DBSession.query(cls)
-            .filter(
-                cls.chambre == chambre,
-                cls.texte == texte,
-                cls.partie == partie,
-                cls.organe == organe,
-            )
-            .exists()
-        ).scalar()
-        return res
+        lecture = cls._get_helper(
```

Comment:
The previous code was only checking the existence, not sure if it may impact performances.

Reply:
I refactored this part a bit, you might like it better that way.